File: 9. Find Immediate Greater Than X - GFG/9.-find-immediate-greater-than-x.cpp
// { Driver Code Starts
//Initial Template for C++

#include<bits/stdc++.h>
using namespace std;

 // } Driver Code Ends
//User function Template for C++


class Solution{
  public:
    int immediateGreater(int arr[], int n, int x){
        
        int closest = 10001;
        
        for(int i = 0; i < n; i++){
            if( arr[i] > x && arr[i] < closest){
                closest = arr[i];
            }
            
        }
        if (closest == 10001 ){ 
        return -1;
        }
        else return closest;
        }
};

// { Driver Code Starts.



int main() {
	int t;
	cin>>t;
	while(t--)
	{
    	int n;
    	cin >> n;
    	
    	int arr[n];
    	
    	for(int i = 0;i<n;i++){
    	    cin >> arr[i];
    	}
    	
    	int x;
    	cin >> x;
    	Solution obj;
    	cout << obj.immediateGreater(arr, n, x) << endl;
	}
	
	return 0;
}  // } Driver Code Ends

File: Array/9. Find Immediate Greater Than X - GFG/9.-find-immediate-greater-than-x.py
#User function Template for python3

class Solution:
    # inf has been imported in driver code
    def immediateGreater(self,arr,n,x):
        #return required ans
        
        #code here
        closest = 10001
        for i in range(n):
            if arr[i] > x and arr[i] < closest:
                closest = arr[i]
        
        if closest == 10001:
            return -1
        else: 
            return closest

#{ 
#  Driver Code Starts
#Initial Template for Python 3

#contributed by RavinderSinghPB
if __name__ =='__main__':
    from math import inf
    tcs=int(input())
    
    for _ in range(tcs):
        n=int(input())
        arr=[int(e) for e in input().split()]
        x=int(input())
        
        ans=Solution().immediateGreater(arr,n,x)
        print(ans)
# } Driver Code Ends